cinema/views.py

```diff
@@ -43,10 +45,42 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        actors = self.request.query_params.get("actors")
+        genres = self.request.query_params.get("genres")
+        title = self.request.query_params.get("title")
+        queryset = self.queryset
+
+        if actors:
+            actors_ids = [int(str_id) for str_id in actors.split(",")]
+            queryset = Movie.objects.filter(actors__id__in=actors_ids)
+
+        if genres:
+            genres_ids = [int(str_id) for str_id in genres.split(",")]
+            queryset = Movie.objects.filter(genres__id__in=genres_ids)
+
+        if title:
+            queryset = Movie.objects.filter(title__icontains=title)
+
+        if self.action == ("list", "retrieve"):
+            queryset = Movie.objects.prefetch_related("actors")
```

Comment:
The condition `if self.action == ('list', 'retrieve'):` seems incorrect. The `self.action` attribute will not match a tuple of strings. Consider using `if self.action in ['list', 'retrieve']:` to check if the action is either 'list' or 'retrieve'.